src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *     security:
+ *       - BearerAuth: []  # Indicates that the endpoint requires a bearer token for authentication
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               title:
+ *                 type: string
+ *                 description: The title of the post.
+ *               content:
+ *                 type: string
+ *                 description: The content of the post.
+ *               category_url:
+ *                 type: string
+ *                 description: The category of the post.
+ *               price:
+ *                 type: number
+ *                 description: The price associated with the post (if applicable).
+ *             required:
+ *               - title
+ *               - content
+ *     responses:
+ *       201:
+ *         description: The newly created post.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 _id:
+ *                   type: string
+ *                   description: The unique identifier of the post.
+ *                 title:
+ *                   type: string
+ *                   description: The title of the post.
+ *                 content:
+ *                   type: string
+ *                   description: The content of the post.
+ *                 category_url:
+ *                   type: string
+ *                   description: The category of the post.
+ *                 price:
+ *                   type: number
+ *                   description: The price associated with the post (if applicable).
+ *                 createdAt:
+ *                   type: string
+ *                   format: date-time
+ *                   description: The creation timestamp of the post.
+ *                 creator_id:
+ *                   type: object
+ *                   properties:
+ *                     _id:
+ *                       type: string
+ *                       description: The unique identifier of the creator.
+ *                     username:
+ *                       type: string
+ *                       description: The username of the post creator.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message.
+ */
 router.post('/', auth, postControl, uploadPost);
+
+/**
+ * @swagger
+ * /posts/{postID}/like:
+ *   post:
+ *     tags:
+ *       - Posts operations
+ *     summary: Like or unlike a post , Must be connected as user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"